sdk/python/kfp/compiler/compiler_utils.py

```diff
@@ -777,15 +777,16 @@ def get_dependencies(
     return dependencies
 
 
-def recursive_replace_placeholders(data: Union[Dict, List], old_value: str,
-                                   new_value: str) -> Union[Dict, List]:
-    """Recursively replaces values in a nested dict/list object.
+def recursive_replace_placeholders(data: Any, old_value: str,
```

Comment:
What happens if data is neither a dict, list, str, nor a pipelinechannel?

Reply:
Thank you for your review!

It returns the original data itself, if data is neither a dict, list, str nor a pipelinechannel.

The purpose of the function is to replace the parameters in the original kfp v1 format (e.g. `{{inputs.parameters.parameter_name}}`) to kfp v2 format (e.g. `{{$.inputs.parameters[pipelinechannel--parameter_name-Output]}}`). So, the function has to handle `str` values (including those which in `dict` and `list`)

Reply:
@droctothorpe Please let me know if there is anything remains unclear.